packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -364,32 +397,58 @@ namespace Ifpack2 {
       /// use scalar_type for communication data type enum
       using scalar_type = typename impl_type::scalar_type;
 
-      template <typename T>
-      static int isend(const MPI_Comm comm, const T* buf, int count, int dest, int tag, MPI_Request* ireq) {
+      static int isend(const MPI_Comm comm, const char* buf, int count, int dest, int tag, MPI_Request* ireq) {
```

Comment:
For the future: Do we actually need this?  It looks like you just call `MPI_Isend` in some places, and in others, you call this `isend` function.

Reply:
I don't think that I use send/recv other places. Communication happens in AsyncableImporter (Send/Recv) and NormManager (AllReduce).

Reply:
It's OK, I'm not so worried about this.  It could lead to a multiple definition error if ETI is OFF, though.